### DISCUSSION TITLE
First impressions from Neovim user (and what currently stops me from switching)

### DISCUSSION THREAD

**Discussion by psiho** [zed-industries/zed#18125] [Feature Requests]
In short... very promising. Really amazing work done here and my impression is that all we need is people starting work on extensions, basically copying VSCode and Neovim ones. Since this is written in Rust, I think it will not be long now :)
I love the amount of attention given to vim mode and keyboard shortcuts configuration.

Now about **elephant in (my) room**:  Telescope (or Fzf) functionality. Yes, multi buffers in Zed's native search can be useful, but Telescope is so much more. Searching (with preview) of file names, text across files and buffers (tabs) is a minimum for me. But Telescope/Fzf can search almost anything you pass to it.

**What hurts but I could live without it** (probably):
- Terminal... most of us have heavily configured terminal choices (Kitty, Alacrity) or use Tmux. I don't see it possible that any amount of work put into internal terminal will help here. I'll always prefer my Kitty with splits, tabs, scripts, etc. But with tiling window manager, and Zed's tasks, this can work, with some changes in my workflow and using tiling window manager advantages.
- QwickFix list - what to say. A horizontal split window that can be fed with any list output (file search, text search, diagnostics, todo lists,...)

Really, can't wait for extensions to pick up. And related to that... maybe it's already time to categorize them on website and inside the editor. It's a pain to scroll through themes to see whats new there

**Comment by ScriptAlchemist** [comment]
I'm going to agree with some of those things as well. What slightly forced my hand is the agentic features paired with a more vim focused development environment.

So even though it's not perfect. Since progression in my job required me to have a more built out agent setup. That I couldn't really get with nvim unless I spent days/weeks configuring. Jumping into zed has been pleasant to say the least.

I did add a few tools that pull up some terminal windows to use git, monitoring and other tools that zed doesn't have access to, but it's nice to be able to reach out.

Honestly. I can still use nvim inside of zed. When the tooling that is there. Like pandoc. Isn't intergrated the same way.

I did need to fix the navigation between panels. To have it more like tmux. Where I can jump around with ctrl a + hjkl

So if you like Nvim and you need an agent setup. I would suggest giving zed a shot for sure.

Added below

```
[
  {
    "context": "Editor",
    "bindings": {
      // Panel splitting (similar to screen bindings)
      "ctrl-a \\": "pane::SplitRight",
      "ctrl-a -": "pane::SplitDown",
      // Navigation between panes (vim-style with ctrl-a prefix)
      "ctrl-a h": "workspace::ActivatePaneLeft",
      "ctrl-a j": "workspace::ActivatePaneDown",
      "ctrl-a k": "workspace::ActivatePaneUp",
      "ctrl-a l": "workspace::ActivatePaneRight",
      // Resizing panes (similar to tmux)
      "ctrl-a ctrl-h": "vim::ResizePaneLeft",
      "ctrl-a ctrl-j": "vim::ResizePaneDown",
      "ctrl-a ctrl-k": "vim::ResizePaneUp",
      "ctrl-a ctrl-l": "vim::ResizePaneRight",
      "alt-d": "editor::ToggleBreakpoint"
    }
  },
  {
    "context": "Workspace",
    "bindings": {
      "ctrl-a ctrl-c": "file_finder::Toggle",
      "ctrl-a f": "file_finder::Toggle",
      "ctrl-a t": "terminal_panel::ToggleFocus",
    }
  },
  {
    "context": "Terminal",
    "bindings": {
      // Navigation between panes (vim-style with ctrl-a prefix)
      "ctrl-a h": "workspace::ActivatePaneLeft",
      "ctrl-a j": "workspace::ActivatePaneDown",
      "ctrl-a k": "workspace::ActivatePaneUp",
      "ctrl-a l": "workspace::ActivatePaneRight"
    }
  },
  {
    "context": "Editor && (showing_code_actions)",
    "bindings": {
      "k": "editor::ContextMenuPrevious",
      "j": "editor::ContextMenuNext"
    }
  },
  {
    "context": "Editor && (showing_completions)",
    "bindings": {
      "ctrl-k": "editor::ContextMenuPrevious",
      "ctrl-j": "editor::ContextMenuNext"
    }
  },
  {
    "context": "Picker > Editor",
    "bindings": {
      "ctrl-k": "menu::SelectPrevious",
      "ctrl-j": "menu::SelectNext"
    }
  },
  {
    "context": "ProjectPanel && not_editing",
    "bindings": {
      // Navigation between panes (vim-style with ctrl-a prefix)
      "ctrl-a h": "workspace::ActivatePaneLeft",
      "ctrl-a j": "workspace::ActivatePaneDown",
      "ctrl-a k": "workspace::ActivatePaneUp",
      "ctrl-a l": "workspace::ActivatePaneRight"
    }
  }
]
```

**Comment by lungarella-raffaele** [comment]
> * QwickFix list - what to say. A horizontal split window that can be fed with any list output (file search, text search, diagnostics, todo lists,...)

Absolutely a fan of this. The problem pane is not enough for me. I often use external scripts to check my codebase, would be awesome to write a task that runs this scripts and dumps the result in a multibuffer view

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
